File: JULY-2025/15-07-2025/p3136.php
class Solution {

    /**
     * @param String $word
     * @return Boolean
     */
    function isValid($word) {
        $vowels = "aeiouAEIOU";
        $containsCons = false;
        if(strlen($word) < 3) return false;
        for($i = 0; $i<strlen($word); $i++){
            $c = ord($word[$i]);
            if(($c >= 65 && $c <= 90) || ($c >= 97 && $c <= 122)){
               
                if(str_contains($vowels, chr($c))){
                    $vowels .= '+';
                }else{
                    $containsCons = true;
                }
            }else if($c >= 48 && $c <= 57){
                 continue;
            }else{
                return false;
            }
        }
       
        return (str_contains($vowels, '+') && $containsCons);
    }
}

File: JULY-2025/15-07-2025/p3136.swift
// characters ascii codes
let lowercaseCodes = UInt8(97)...UInt8(122) // a-z
let uppercaseCodes = UInt8(65)...UInt8(90)  // A-Z
let digitCodes = UInt8(48)...UInt8(57)      // 0-9
let vowelCodes: Set<UInt8> = [65,69,73,79,85,97,101,105,111,117] // aeiouAEIOU

class Solution {
    func isValid(_ word: String) -> Bool
    {
        if word.utf8.count < 3 { return false }

        var hasVowels = false, hasConsonants = false
        for c in word.utf8
        {
            if vowelCodes.contains(c) {
                hasVowels = true
            }
            else if uppercaseCodes.contains(c) || lowercaseCodes.contains(c) {
                hasConsonants = true
            }
            else if !digitCodes.contains(c) {
                return false
            }
        }
        return hasVowels && hasConsonants
    }
}